sample-apps/python-manual-instrumentation-sample-app/app.py

```diff
@@ -0,0 +1,220 @@
+from config import *
+
+import random_metrics
+import requests
+import random
+import boto3
+
+from opentelemetry import propagate, trace, metrics
+
+from flask import Flask, request
+
+# Exporter Libraries
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+
+# Metrics Library
+from opentelemetry.metrics import CallbackOptions, Observation
+
+# SDK Libraries
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.sdk.extension.aws.trace import AwsXRayIdGenerator
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.exporter.otlp.proto.grpc.metric_exporter import OTLPMetricExporter
+
+# Propogators Library
+from opentelemetry.propagators.aws import AwsXRayPropagator
+from opentelemetry.propagators.aws.aws_xray_propagator import (
+    TRACE_ID_DELIMITER,
+    TRACE_ID_FIRST_PART_LENGTH,
+    TRACE_ID_VERSION,
+)
+
+# Instrumentation Libraries
+from opentelemetry.instrumentation.botocore import BotocoreInstrumentor
+from opentelemetry.instrumentation.flask import FlaskInstrumentor
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+
+
+# Set up AWS X-Ray Propagator
+propagate.set_global_textmap(AwsXRayPropagator())
+
+# Service name is required for most backends
+resource = Resource(attributes={
+    SERVICE_NAME: 'python-manual-instrumentation-sampleapp'
+})
+
+# Setting up Traces
+processor = BatchSpanProcessor(OTLPSpanExporter())
+tracer_provider = TracerProvider(
+    resource=resource, 
+    active_span_processor=processor,
+    id_generator=AwsXRayIdGenerator())
+
+trace.set_tracer_provider(tracer_provider)
+tracer = trace.get_tracer(__name__)
+
+# Setting up Metrics
+metric_reader = PeriodicExportingMetricReader(OTLPMetricExporter())
+metric_provider = MeterProvider(resource=resource, metric_readers=[metric_reader])
+
+metrics.set_meter_provider(metric_provider)
+meter = metrics.get_meter(__name__)
+
+# Starting Flask app
+app = Flask(__name__)
+
+# Instrument Packages
+
+BotocoreInstrumentor().instrument()
+FlaskInstrumentor().instrument_app(app)
+RequestsInstrumentor().instrument()
+
+# Generate config from config file
+cfg = create_config('config.yaml')
```

Comment:
nit: I would wrap these in functions. 

eg.:
```python
def setup_opentelemetry(): 
   processor = BatchSpanProcessor(OTLPSpanExporter())
   tracer_provider = TracerProvider(
        resource=resource, 
        active_span_processor=processor,
        id_generator=AwsXRayIdGenerator())
  
    trace.set_tracer_provider(tracer_provider)
    tracer = trace.get_tracer(__name__)
   
    # Setting up Metrics
    metric_reader = PeriodicExportingMetricReader(OTLPMetricExporter())
    metric_provider = MeterProvider(resource=resource, metric_readers=[metric_reader])

    metrics.set_meter_provider(metric_provider)
    meter = metrics.get_meter(__name__)

def setup_instrumentation():
    BotocoreInstrumentor().instrument()
    FlaskInstrumentor().instrument_app(app)
    RequestsInstrumentor().instrument()

```

and call those in `if __name__ == '__main__':`

The idea is that you can replace lots of your comments by your functions. Also this an anti-pattern in python. If this `app.py` was imported somewhere else, a lot of code would be executed in import time.